packages/core/src/eventUtils.ts

```diff
@@ -0,0 +1,72 @@
+import { XSTATE_INIT } from './constants.ts';
+import {
+  DoneInvokeEventObject,
+  DoneStateEventObject,
+  ErrorPlatformEvent
+} from './types.ts';
+import * as constantPrefixes from './constantPrefixes.ts';
+
+/**
+ * Returns an event type that represents an implicit event that
+ * is sent after the specified `delay`.
+ *
+ * @param delayRef The delay in milliseconds
+ * @param id The state node ID where this event is handled
+ */
+export function after(delayRef: number | string, id?: string) {
+  const idSuffix = id ? `#${id}` : '';
+  return `${constantPrefixes.after}(${delayRef})${idSuffix}`;
+}
+
+export function doneStateEventType<T extends string = string>(id: T) {
```

Comment:
let's bikeshed how those helper functions should be named (how to call the utils responsible for creating the event type and how do we call utils responsible for creating the event itself). We also need to decide which ones should be publicly exported (if any)

Reply:
By default, let's not export any

Reply:
I pushed out extra changes, could you re-review?